design-document.md

```diff
@@ -106,6 +123,167 @@ to achieve these goals:
+1. Relaying messages between aggregators. The leader can compromise availability
+   by dropping messages.
+
+##### Mitigations
+
+1. Rather than having all aggregators trust the leader to validate proofs, each
+   aggregator could exchange proof shares with every other aggregator, enabling
+   each aggregator to independently verify the proofs.
+1. Running the protocol over multiple subsets of the available aggregators and
+   choosing different leaders for each run would prevent any individual leader
+   from compromising availability.
+
+#### Collector
+
+##### Capabilities
+
+1. Advertising shared configuration parameters (e.g., minimum thresholds for
+   aggregations, polynomial identity test values, arithmetic circuits).
+1. Collectors may trivially defeat availability by discarding aggregation
+   parts submitted by aggregators.
+
+##### Mitigations
+
+1. Aggregators should refuse shared parameters that are trivially insecure
+   (i.e., aggregation threshold of 1 contribution).
+
+#### Aggregator collusion
+
+If all aggregators collude (e.g. by promiscuously sharing unencrypted data
+shares), then none of the properties of the system hold. Accordingly, such
+scenarios are outside of the threat model.
+
+#### Attacker on the network
+
+We assume the existence of attackers on the network links between participants.
+
+##### Capabilities
+
+1. Observation of network traffic. Attackers may observe messages exchanged
+   between participants at the IP layer.
+     1. The time of transmission of data shares by clients could reveal
+        information about user activity.
+          * For example, if a user opts into a new feature, and the client
+            immediately reports this to aggregators, then just by observing
+            network traffic, the attacker can infer what the user did.
+     1. Observation of message size could allow the attacker to learn how much
+        data is being submitted by a client.
+          * For example, if the attacker observes an encrypted message of some
+            size, they can infer the size of the plaintext, plus or minus the
+            cipher block size. From this they may be able to infer which
+            aggregations the user has opted into or out of.
+1. Tampering with network traffic. Attackers may drop messages or inject new
+   messages into communications between participants.
+
+##### Mitigations
+
+1. All messages exchanged between participants in the system should be
+   encrypted.
+1. All messages exchanged between aggregators, the collector and the leader
+   should be mutually authenticated so that network attackers cannot impersonate
+   participants.
+1. Clients should be required to submit data at regular intervals so that the
+   timing of individual messages does not reveal anything.
+1. A fixed-length encoding should be used for data shares. Additionally, clients
+   should submit dummy data even for aggregations the user has not opted into.
```

Comment:
Aggregation isn't well-defined if input lengths vary, so I think the first part of this mitigation is irrelevant.